src/shogun/lib/SGStringList.cpp

```diff
@@ -14,7 +14,7 @@ SGStringList<T>::SGStringList() : SGReferencedData()
 }
 
 template <class T>
-SGStringList<T>::SGStringList(SGString<T>* s, index_t num_s, index_t max_length,
+SGStringList<T>::SGStringList(SGVector<T>* s, index_t num_s, index_t max_length,
```

Comment:
long term...this structure should be replaced with std::vector .. but maybe not just yet to keep the PR smaller?

Reply:
yes. also we need to get rid of all of these SGString pointer arrays and replace them either with std::vector or SGStringList.
the problem with std::vector is that they're expensive to copy, and can't be shared. in Qt for example there is QStringList which does intrusive reference counting like we do here.

Reply:
I noticed that pointers of SGString are passed around alot, so I assumed they're shared between classes, I will check.

Reply:
I think they are.... updates here?
the std::vector caused problems in the end I assume?